<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>871</width>
    <height>633</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Lab3</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <layout class="QVBoxLayout" name="verticalLayout" stretch="1,0">
      <item>
       <widget class="GLWidget" name="openGLWidget">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>533</width>
          <height>533</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>533</width>
          <height>533</height>
         </size>
        </property>
        <property name="cursor">
         <cursorShape>CrossCursor</cursorShape>
        </property>
        <property name="mouseTracking">
         <bool>true</bool>
        </property>
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSplitter" name="splitter">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <widget class="QWidget" name="horizontalLayoutWidget">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="rgb_label">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>Red = x Green = y Blue = z</string>
            </property>
            <property name="buddy">
             <cstring>openGLWidget</cstring>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QGridLayout" name="gridLayout_3">
        <property name="sizeConstraint">
         <enum>QLayout::SetFixedSize</enum>
        </property>
        <item row="6" column="1">
         <widget class="AngleSpinBox" name="y_box">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>26</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="locale">
           <locale language="English" country="UnitedStates"/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
          <property name="prefix">
           <string/>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
          <property name="minimum">
           <double>-72000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>72000.000000000000000</double>
          </property>
          <property name="value">
           <double>0.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>Initial state</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>x angle</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="buddy">
           <cstring>x_box</cstring>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="pushButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string>Change colors</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="scale_box">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>97</width>
            <height>26</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
          <property name="locale">
           <locale language="English" country="UnitedStates"/>
          </property>
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
          <property name="prefix">
           <string/>
          </property>
          <property name="maximum">
           <double>1000.000000000000000</double>
          </property>
          <property name="value">
           <double>100.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="6" column="2">
         <widget class="AngleSpinBox" name="z_box">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>26</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>100</width>
            <height>27</height>
           </size>
          </property>
          <property name="locale">
           <locale language="English" country="UnitedStates"/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
          <property name="prefix">
           <string/>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
          <property name="minimum">
           <double>-10000000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>10000000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="0" colspan="3">
         <widget class="Line" name="line">
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="5" column="2">
         <widget class="QLabel" name="label_6">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>z angle</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="buddy">
           <cstring>z_box</cstring>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="AngleSpinBox" name="x_box">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>26</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="locale">
           <locale language="English" country="UnitedStates"/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
          <property name="prefix">
           <string/>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
          <property name="minimum">
           <double>-72000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>72000.000000000000000</double>
          </property>
          <property name="value">
           <double>0.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="3">
         <widget class="Line" name="line_3">
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QCheckBox" name="checkBox_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Edges</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Approx:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>approx</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QCheckBox" name="checkBox_3">
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Base</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QSpinBox" name="approx">
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="checkBox">
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Color</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <property name="tristate">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="pushButton_3">
          <property name="text">
           <string>Auto scale</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>y angle</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="buddy">
           <cstring>y_box</cstring>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_7">
          <property name="layoutDirection">
           <enum>Qt::RightToLeft</enum>
          </property>
          <property name="text">
           <string>Scale:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>scale_box</cstring>
          </property>
         </widget>
        </item>
        <item row="7" column="0" colspan="3">
         <widget class="Line" name="line_2">
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>871</width>
     <height>27</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>GLWidget.hpp</header>
   <slots>
    <signal>scale_changed(double)</signal>
    <signal>scale_message(QString)</signal>
    <signal>angle_changed(Vector3f)</signal>
    <slot>set_scale(double)</slot>
    <slot>color(bool)</slot>
    <slot>edges(bool)</slot>
    <slot>base(bool)</slot>
    <slot>initial_state()</slot>
    <slot>auto_scale()</slot>
    <slot>change_colors()</slot>
    <slot>set_angle(Vector3f,char)</slot>
    <slot>set_approx(int)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>AngleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>AngleSpinBox.hpp</header>
   <slots>
    <signal>value_changed(Vector3f,char)</signal>
    <slot>set_value(Vector3f)</slot>
    <slot>set_value(double)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>approx</tabstop>
  <tabstop>checkBox_2</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>x_box</tabstop>
  <tabstop>pushButton_3</tabstop>
  <tabstop>y_box</tabstop>
  <tabstop>pushButton_2</tabstop>
  <tabstop>z_box</tabstop>
  <tabstop>scale_box</tabstop>
  <tabstop>checkBox_3</tabstop>
  <tabstop>checkBox</tabstop>
  <tabstop>openGLWidget</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>checkBox_2</sender>
   <signal>clicked(bool)</signal>
   <receiver>openGLWidget</receiver>
   <slot>edges(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>270</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>379</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>openGLWidget</receiver>
   <slot>color(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>177</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>429</x>
     <y>63</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_3</sender>
   <signal>clicked(bool)</signal>
   <receiver>openGLWidget</receiver>
   <slot>base(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>66</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>445</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>initial_state()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>41</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>402</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>auto_scale()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>390</x>
     <y>129</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>change_colors()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>253</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>376</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scale_box</sender>
   <signal>valueChanged(double)</signal>
   <receiver>openGLWidget</receiver>
   <slot>set_scale(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>402</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x_box</sender>
   <signal>value_changed(Vector3f,char)</signal>
   <receiver>openGLWidget</receiver>
   <slot>set_angle(Vector3f,char)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>65</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>280</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>y_box</sender>
   <signal>value_changed(Vector3f,char)</signal>
   <receiver>openGLWidget</receiver>
   <slot>set_angle(Vector3f,char)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>331</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>z_box</sender>
   <signal>value_changed(Vector3f,char)</signal>
   <receiver>openGLWidget</receiver>
   <slot>set_angle(Vector3f,char)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>angle_changed(Vector3f)</signal>
   <receiver>z_box</receiver>
   <slot>set_value(Vector3f)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>495</x>
     <y>407</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>angle_changed(Vector3f)</signal>
   <receiver>y_box</receiver>
   <slot>set_value(Vector3f)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>350</x>
     <y>437</y>
    </hint>
    <hint type="destinationlabel">
     <x>155</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>angle_changed(Vector3f)</signal>
   <receiver>x_box</receiver>
   <slot>set_value(Vector3f)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>340</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>50</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>approx</sender>
   <signal>valueChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>set_approx(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>193</y>
    </hint>
    <hint type="destinationlabel">
     <x>362</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x_box</sender>
   <signal>valueChanged(double)</signal>
   <receiver>x_box</receiver>
   <slot>set_value(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>33</x>
     <y>161</y>
    </hint>
    <hint type="destinationlabel">
     <x>72</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>y_box</sender>
   <signal>valueChanged(double)</signal>
   <receiver>y_box</receiver>
   <slot>set_value(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>z_box</sender>
   <signal>valueChanged(double)</signal>
   <receiver>z_box</receiver>
   <slot>set_value(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>278</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
